source/class/qx/ui/form/FileSelectorButton.js

```diff
@@ -0,0 +1,154 @@
+ * let button = new qx.ui.form.FileSelectorButton("Select File");
+ * button.addListener('changeFileSelection',function(e){
+ * let fileList = e.getData();
+ *   let form = new FormData();
+ *   form.append('file',fileList[0]);
+ *   let req = new qx.io.request.Xhr("upload",'POST').set({
+ *     requestData: form
+ *   });
+ *   req.addListener('success',(e) => {
+ *     let response = req.getResponse();
+ *   });
+ * });
+ * ```
+ * 
+ * Process the file directly in javascript using the FileReader API.
+ * 
+ * ```javascript
+ * let button = new qx.ui.form.FileSelectorButton("Select File");
+ * button.addListener('changeFileSelection',function(e){
+ *   let fileList = e.getData();
+ *   const reader = new FileReader();
+ *   reader.addEventListener('load', () => {
+ *     let response = reader.result;
+ *     console.log("The first 4 chrs are: " + response);
+ *   });
+ *   const file = fileList[0];
+ *   reader.readAsText(file.slice(0,4));
+ * });
+
+*
+ * [MDN Documentation](https://developer.mozilla.org/en-US/docs/Web/API/File_API/Using_files_from_web_applications)
+ */
+
+let InputElementIdCounter = 0;
+qx.Class.define("qx.ui.form.FileSelectorButton", {
+  extend: qx.ui.form.Button,
+  events: {
+    /**
+     * The event is fired when the file selection changes.
+     *
+     * The method {@link qx.event.type.Data#getData} returns the
+     * current [fileList](https://developer.mozilla.org/en-US/docs/Web/API/FileList)
+     */
+    changeFileSelection: "qx.event.type.Data"
+  },
+  properties: {
+    /**
+     * What type of files should be offered in the fileselection dialog.
+     * Use a comma separated list of [Unique file type specifiers](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#unique_file_type_specifiers). If you dont set anything, all files
+     * are allowed.
+     *
+     * *Example*
+     *
+     * `.doc,.docx,application/msword`
+     */
+    accept: {
+      nullable: true,
+      check: "String",
+      apply: "_applyAttribute"
+    },
+    /**
+     * Specify that the camera should be used for getting the "file". The
+     * value defines which camera should be used for capturing images.
+     * `user` indicates the user-facing camera.
+     * `environment` indicates the camera facing away from the user.
+     */
+    capture: {
+      nullable: true,
+      check: ["user", "environment"],
+      apply: "_applyAttribute"
+    },
+    /**
+     * Set to "true" if you want to allow the selection of multiple files.
+     */
+    multiple: {
+      nullable: true,
+      check: "Boolean",
+      apply: "_applyAttribute"
+    },
+    /**
+     * If present, indicates that only directories should be available for
+     * selection.
+     */
+    directoriesOnly: {
+      nullable: true,
+      check: "Boolean",
+      apply: "_applyAttribute"
+    }
+  },
+  members: {
+    __inputObject: null,
+
+    _applyAttribute: function (value, old, attr) {
+      if (attr === "directoriesOnly") {
```

Comment:
Sorry. This is good, but the comment you had about it applying other than just to webkit was good. Can we get that comment back, here?

Reply:
sure !